pinot-connectors/pinot-spark-connector/src/main/scala/org/apache/pinot/connector/spark/connector/PinotGrpcServerDataFetcher.scala

```diff
@@ -25,14 +25,15 @@ import org.apache.pinot.common.proto.Server.ServerRequest
 import org.apache.pinot.connector.spark.utils.Logging
 import org.apache.pinot.spi.config.table.TableType
 
+import java.io.Closeable
 import scala.collection.JavaConverters._
 
 /**
  * Data fetcher from Pinot Grpc server with specific segments.
  * Eg: offline-server1: segment1, segment2, segment3
  */
 private[pinot] class PinotGrpcServerDataFetcher(pinotSplit: PinotSplit)
-  extends Logging {
+  extends Logging with Closeable {
```

Comment:
it might be eaiser to move channel/stub inside fetchData method as they are used once and are not globally shared

Reply:
Thanks for the review @walterddr!

Do you mean closing the channel inside the `fetchData` method? If so, that may not be possible. We need to return a Closeable iterator to Spark which will do the closing once the stream is fully consumed.

I guess we could return an anonymous function which handles the closure, but I thought implementing 'Closeable' makes it very obvious to the user.

Let me know what you think.

Reply:
ah. I see. previously it was returning a List so thus we can close directly when fetchData returns. great catch.